packages/realm-server/tests/realm-endpoints/lint-test.ts

```diff
@@ -42,130 +54,821 @@ module(`realm-endpoints/${basename(__filename)}`, function () {
+          inferPrettierParser('test.ts'),
+          'typescript',
+          'TS files use typescript parser',
+        );
+        assert.strictEqual(
+          inferPrettierParser('test.js'),
+          'babel',
+          'JS files use babel parser',
+        );
+        assert.strictEqual(
+          inferPrettierParser('unknown.ext'),
+          'glimmer',
+          'Unknown extensions default to glimmer',
+        );
+      });
+
+      test('basic prettier formatting works on GTS content', async function (assert) {
+        try {
+          const prettier = await import('prettier');
 
-      assert.strictEqual(response.status, 401, 'HTTP 401 status');
+          const input = `import{CardDef}from 'somewhere';export class MyCard extends CardDef{@field name=contains(StringField);}`;
+          const config = {
+            singleQuote: true,
+            plugins: ['prettier-plugin-ember-template-tag'],
+            parser: 'glimmer',
+          };
+
+          const result = await prettier.format(input, config);
+
+          assert.ok(result, 'Prettier formatting produces output');
+          assert.strictEqual(
+            typeof result,
+            'string',
+            'Prettier result is a string',
+          );
+          assert.ok(
+            result.includes('import'),
+            'Import statements are preserved',
+          );
+          assert.ok(result.includes('CardDef'), 'CardDef is preserved');
+          assert.ok(
+            result.includes('export'),
+            'Export statements are preserved',
+          );
+
+          // Check that formatting actually improves the code structure
+          // The input is minified, so formatting should add whitespace
+          const hasProperSpacing =
+            result.includes('import {') ||
+            result.includes("from '") ||
+            result.includes('export ');
+          assert.ok(
+            hasProperSpacing,
+            `Formatted output has proper spacing. Input: ${input.length} chars, Output: ${result.length} chars. Result: ${result.substring(0, 100)}...`,
+          );
+        } catch (error) {
+          assert.ok(
+            false,
+            `Basic prettier formatting failed: ${(error as Error).message}`,
+          );
+        }
+      });
+
+      test('performance benchmarking works', async function (assert) {
+        const benchmark = await benchmarkOperation(
+          'test-operation',
+          () => new Promise((resolve) => setTimeout(resolve, 1)),
+          10,
+        );
+
+        assert.ok(benchmark.duration > 0, 'Benchmark should record duration');
+        assert.strictEqual(
+          benchmark.iterations,
+          10,
+          'Should run correct number of iterations',
+        );
+        assert.ok(benchmark.averageTime > 0, 'Should calculate average time');
+        assert.ok(
+          benchmark.maxTime >= benchmark.minTime,
+          'Max time should be >= min time',
+        );
+      });
```

Comment:
I still don’t understand the purpose of this benchmarking. If we want this, is it really Prettier-specific? And what performance are we expecting?